web/src/unread.ts

```diff
@@ -964,6 +964,17 @@ export function get_msg_ids_for_topic(stream_id: number, topic_name: string): nu
     return unread_topic_counter.get_msg_ids_for_topic(stream_id, topic_name);
 }
 
+export function dms_has_mention(msg_ids: number | number[]): boolean {
+    const idsToCheck = Array.isArray(msg_ids) ? msg_ids : [msg_ids];
+
+    for (const msg_id of idsToCheck) {
+        if (message_has_mention(msg_id)) {
+            return true;
+        }
+    }
+    return false;
+}
+
```

Comment:
Architecturally, I'm not sure about the design here; it seems like it might be cleaner to have this code path compute which DM conversations contain mentions when processing everything in `unread.ts`, rather than having these follow-up calls, each of which do a nontrivial loop:

```
    get_counts(include_latest_msg_id = false): DirectMessageCountInfo {                                 
        const pm_dict = new Map(); // Hash by user_ids_string -> count Optional[, max_id]               
        let total_count = 0;                                                                            
        for (const [user_ids_string, id_set] of this.bucketer) {                                        
            const count = id_set.size;                                                                  
            if (include_latest_msg_id) {                                                                
                const latest_msg_id = Math.max(...id_set);                                              
                pm_dict.set(user_ids_string, {                                                          
                    count,                                                                              
                    latest_msg_id,                                                                      
                });                                                                                     
            } else {                                                                                    
                pm_dict.set(user_ids_string, count);                                                    
            }                                                                                           
            total_count += count;                                                                       
        }                                                                                               
        return {                                                                                        
            total_count,                                                                                
            pm_dict,                                                                                    
        };                                                                                              
    }                                                                                                   
```